src/Param.cc

```diff
@@ -613,10 +719,43 @@ bool Param::SetFromString(const std::string &_value)
   return this->dataPtr->set;
 }
 
+//////////////////////////////////////////////////
+bool Param::SetFromString(const std::string &_value)
+{
+  return this->SetFromString(_value, false);
+}
+
+//////////////////////////////////////////////////
+ElementPtr Param::GetParentElement() const
+{
+  return this->dataPtr->parentElement.lock();
+}
+
+//////////////////////////////////////////////////
+void Param::SetParentElement(const ElementPtr _parentElement)
+{
+  if (!_parentElement)
+    this->dataPtr->parentElement.reset();
```

Comment:
nit: I don't think `reset` is necessary here.  `this->dataPtr->parentElement = _parentElement` should do it.

Reply:
Done! https://github.com/ignitionrobotics/sdformat/pull/589/commits/e44c8409affed3c9e75d55b70e1076c29aa7d9f0